### PR TITLE
Assessment use cases

### PR DESCRIPTION
**Pull request by samielsabri**
This pull request includes the first drafts of following use cases (each have their own request, response, InputBoundary, and usecase Implementation):
- AddSimpleAssessment
- SetAssessmentTitle
- SetSimpleWeightScheme
- SetMark
- CommitMark
- UncommitMark
-

### CODE REVIEW

**Comment by DeepSJain** [comment]
Your usecases look good to my understanding, however some of the tests are failing which is probably because you updated some of the entity implementations.

**Review by berrythebear — DISMISSED** [review]

**Review by DeepSJain — APPROVED** [review]
Looks good

**Review by AlanC12138 — APPROVED** [review]
good work, these use cases should be well functioning

**Review by berrythebear — APPROVED** [review]

File: src/main/java/usecases/assessment/SetSimpleWeightSchemeUseCase.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        if (request.weightOfEachInstance > 1) {
+            throw new SetSimpleWeightSchemeInputBoundary.SetSimpleWeightSchemeError("Weight of each instance must be less than or equal to 1");
+        }
+        if (request.numberOfInstances * request.weightOfEachInstance > 1) {
+            throw new SetSimpleWeightSchemeInputBoundary.SetSimpleWeightSchemeError("Total weight of instances must be less than or equal to 1");
+        }
+
+        SimpleWeight weightscheme = new SimpleWeight(new Weight(request.numberOfInstances, request.weightOfEachInstance));
+
+        assessment.setWeightScheme(weightscheme);
```
Comment (MrSunshineVS) [inline src/main/java/usecases/assessment/SetSimpleWeightSchemeUseCase.java:61]: Need to check whether the assessment contains more submitted marks than the weightscheme's total number of marks, and truncate accordingly. Or throw an error.

**Pull request merged**